### PR TITLE
dungeon: fix formula to run on Apple Silicon and add license

### PR DESCRIPTION
**Pull request by GOFAI**
It turns out that on Apple Silicon opt_pkgshare expanded into a pathname so long that it overran the column limits in old-timey Fortran and gfortran flamed out as a result. I've tweaked the formula to pass `-ffixed-line-length-none` to gfortran. Also added HPND license.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Could you also suggest this change upstream?

**Comment by GOFAI** [comment]
I'll think about changing the makefile to pass `-ffixed-line-length-none` to gfortran in the next release, but I'm hesitant to make a new release solely to do that as it's unnecessary on any platform for the standard way that dungeon builds.

**Comment by iMichka** [comment]
Can you change the commit text to be: `dungeon: fix formula to run on Apple Silicon and add license`
This is our standard for commit titles.

**Comment by GOFAI** [comment]
I pushed a new commit with the changed commit text. Does that do what you need?

**Comment by GOFAI** [comment]
Checking in on this again. Please let me know what I need to do to get review approval for this.

**Review by SMillerDev — DISMISSED** [review]
Thanks @GOFAI  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by SMillerDev** [comment]
> I pushed a new commit with the changed commit text. Does that do what you need?

Unfortunately no, it needs to be squashed with the old commit. Otherwise the bot can't merge it.

**Comment by GOFAI** [comment]
Apologies if force-pushing wasn't the best way to squash the commits--please tell me what I should've done instead if it wasn't.

**Comment by SMillerDev** [comment]
No, force pushing is the way to go. Thanks @GOFAI !

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**
